### ISSUE TITLE
TypeError: "type" must be a Function.

### ISSUE THREAD

**Issue by msmyers** [cowwoc/requirements.js#25]
```
export function getHeaderValue(event: APIGatewayProxyEventV2, key:string) {
    const string = event.headers[key.toLowerCase()];
    requireThat( string+ 'asd', `header:${key}`).asString()
    return string;
    // return requireThat(event.headers[key.toLowerCase()], `header:${key}`).asString().getActual()
}
```

triggers:

```
TypeError: "type" must be a Function.
Actual : class {
constructor(globalConfiguration, context = /* @__PURE__ */ new Map(), assertionsEnabled, diffEnabled = true, typeToStringConverter = /* @__PURE__ */ new Map()) {
this.globalConfiguration = globalConfiguration;
Objects.assertThatTypeOf(context, "context", "Map");
this.context = context;
if (typeof assertionsEnabled === "undefined")
assertionsEnabled = globalConfiguration.assertionsAreEnabled();
this.assertionsEnabled = assertionsEnabled;
if (typeof diffEnabled === "undefined")
diffEnabled = globalConfiguration.isDiffEnabled();
this.diffEnabled = diffEnabled;
this.typeToStringConverter = typeToStringConverter;
}
assertionsAreEnabled() {
return this.assertionsEnabled;
}
withAssertionsEnabled() {
if (this.assertionsEnabled)
return this;
return new Configuration(this.globalConfiguration, this.context, true, this.diffEnabled, this.typeToStringConverter);
}
withAssertionsDisabled() {
if (!this.assertionsEnabled)
return this;
return new Configuration(this.globalConfiguration, this.context, false, this.diffEnabled, this.typeToStringConverter);
}
isDiffEnabled() {
return this.diffEnabled;
}
withDiff() {
if (this.diffEnabled)
return this;
return new Configuration(this.globalConfiguration, this.context, this.assertionsEnabled, true, this.typeToStringConverter);
}
withoutDiff() {
if (!this.diffEnabled)
return this;
return new Configuration(this.globalConfiguration, this.context, this.assertionsEnabled, false, this.typeToStringConverter);
}
getContext() {
return this.context;
}
putContext(name, value) {
Objects.requireThatStringNotEmpty(name, "name");
const newContext = new Map(this.context);
newContext.set(name, value);
return new Configuration(this.globalConfiguration, newContext, this.assertionsEnabled, this.diffEnabled, this.typeToStringConverter);
}
removeContext(name) {
Objects.requireThatStringNotEmpty(name, "name");
const newContext = new Map(this.context);
if (!newContext.delete(name))
return this;
return new Configuration(this.globalConfiguration, newContext, this.assertionsEnabled, this.diffEnabled, this.typeToStringConverter);
}
convertToString(value) {
const type = Objects.getTypeOf(value);
let converter = this.typeToStringConverter.get(type);
if (!converter)
converter = (value2) => Objects.toString(value2);
return converter(value);
}
withStringConverter(type, converter) {
Objects.assertThatTypeOf(type, "type", "string");
Objects.assertThatTypeOf(converter, "converter", "Function");
const newTypeToStringConverter = new Map(this.typeToStringConverter);
newTypeToStringConverter.set(type, converter);
return new Configuration(this.globalConfiguration, this.context, this.assertionsEnabled, this.diffEnabled, newTypeToStringConverter);
}
withoutStringConverter(type) {
Objects.assertThatTypeOf(type, "type", "string");
const newTypeToStringConverter = new Map(this.typeToStringConverter);
if (!newTypeToStringConverter.delete(type))
return this;
return new Configuration(this.globalConfiguration, this.context, this.assertionsEnabled, this.diffEnabled, newTypeToStringConverter);
}
getGlobalConfiguration() {
return this.globalConfiguration;
}
}
Actual.type: AnonymousFunction
at Function.requireThatInstanceOf (file:///Users/msmyers/Sync/volumes/syncthing/projects/niftmint/niftmint-brandnft-api/packages/sst-api/.sst/artifacts/default-rest-api-mongodb-MyStack-Api-Lambda_GET_-/services/functions/index.js:39183:13)
at Function.assertThatInstanceOf (file:///Users/msmyers/Sync/volumes/syncthing/projects/niftmint/niftmint-brandnft-api/packages/sst-api/.sst/artifacts/default-rest-api-mongodb-MyStack-Api-Lambda_GET_-/services/functions/index.js:39204:25)
at new AbstractObjectValidator (file:///Users/msmyers/Sync/volumes/syncthing/projects/niftmint/niftmint-brandnft-api/packages/sst-api/.sst/artifacts/default-rest-api-mongodb-MyStack-Api-Lambda_GET_-/services/functions/index.js:35191:13)
at new ObjectValidatorImpl (file:///Users/msmyers/Sync/volumes/syncthing/projects/niftmint/niftmint-brandnft-api/packages/sst-api/.sst/artifacts/default-rest-api-mongodb-MyStack-Api-Lambda_GET_-/services/functions/index.js:35715:5)
at Requirements.validateThat (file:///Users/msmyers/Sync/volumes/syncthing/projects/niftmint/niftmint-brandnft-api/packages/sst-api/.sst/artifacts/default-rest-api-mongodb-MyStack-Api-Lambda_GET_-/services/functions/index.js:37706:12)
at Requirements.requireThat (file:///Users/msmyers/Sync/volumes/syncthing/projects/niftmint/niftmint-brandnft-api/packages/sst-api/.sst/artifacts/default-rest-api-mongodb-MyStack-Api-Lambda_GET_-/services/functions/index.js:37696:40)
at requireThat (file:///Users/msmyers/Sync/volumes/syncthing/projects/niftmint/niftmint-brandnft-api/packages/sst-api/.sst/artifacts/default-rest-api-mongodb-MyStack-Api-Lambda_GET_-/services/functions/index.js:40063:20)
at getHeaderValue (file:///Users/msmyers/Sync/volumes/syncthing/projects/niftmint/niftmint-brandnft-api/packages/sst-api/.sst/artifacts/default-rest-api-mongodb-MyStack-Api-Lambda_GET_-/services/functions/index.js:40314:3)
at Runtime.handler (file:///Users/msmyers/Sync/volumes/syncthing/projects/niftmint/niftmint-brandnft-api/packages/sst-api/.sst/artifacts/default-rest-api-mongodb-MyStack-Api-Lambda_GET_-/services/functions/index.js:40319:3)
at Runtime.handleOnce (file:///Users/msmyers/Sync/volumes/syncthing/projects/niftmint/niftmint-brandnft-api/packages/sst-api/node_modules/@serverless-stack/aws-lambda-ric/lib/Runtime/Runtime.js:53:33)
```

**Comment by cowwoc** [comment]
Hi Michael,

From the looks of it, babel (or some other transpiler) is converting functions in an unexpected way.

Can you please clarify whether you are running in node or a browser, and what version you are using?

I'll try to take a look later on this week.

**Comment by msmyers** [comment]
Hi - I'm running locally in NodeJS, though this is an `sst.dev` application, so Lambda (node16, I believe)

**Comment by cowwoc** [comment]
@msmyers Sorry for the super long delay... I ran into technical problems trying to upgrade dependencies (babel being one of them).

I just pushed a commit to `master` that is fully up-to-date. Are you able to repeat your test against it?

If you can't test without an official release pushed to npm then I can publish a new version for you. Let me know.

**Comment by cowwoc** [comment]
Correction, I pushed into https://github.com/cowwoc/requirements.js/tree/version-3.1.0, not the master branch. Please try testing against that version.

**Comment by cowwoc** [comment]
I've published version 3.1.1. Please let me know if this works for you.

**Comment by cowwoc** [comment]
Closing due to inactivity.
